rust/src/domains/mod.rs

```diff
@@ -255,80 +317,33 @@ where
+/// use opendp::domains::{VectorDomain, AtomDomain};
 /// use opendp::core::Domain;
 ///
-/// // Represents the domain of all vectors.
-/// let atom_domain = VectorDomain::new(AtomDomain::new());
-/// assert!(atom_domain.member(&vec![1, 2, 3])?);
+/// // Represents the domain of vectors.
+/// let vec_domain = VectorDomain::new(AtomDomain::default(), None);
+/// assert!(vec_domain.member(&vec![1, 2, 3])?);
 ///
-/// // Represents the domain of all bounded vectors.
-/// let bounded_domain = VectorDomain::new(BoundedDomain::new_closed((-10, 10))?);
+/// // Represents the domain of all vectors of bounded elements.
+/// let bounded_domain = VectorDomain::new(AtomDomain::new_closed((-10, 10))?, None);
 ///
 /// // vec![0] is a member, but vec![12] is not, because 12 is out of bounds of the inner domain
 /// assert!(bounded_domain.member(&vec![0])?);
 /// assert!(!bounded_domain.member(&vec![12])?);
 ///
 /// # opendp::error::Fallible::Ok(())
 /// ```
-#[derive(Clone, PartialEq)]
-pub struct VectorDomain<D: Domain> {
-    pub element_domain: D,
-}
-impl<D: Domain> Debug for VectorDomain<D> {
-    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), std::fmt::Error> {
-        write!(f, "VectorDomain({:?})", self.element_domain)
-    }
-}
-impl<D: Domain + Default> Default for VectorDomain<D> {
-    fn default() -> Self {
-        Self::new(D::default())
-    }
-}
-impl<D: Domain> VectorDomain<D> {
-    pub fn new(element_domain: D) -> Self {
-        VectorDomain { element_domain }
-    }
-}
-impl<T: CheckNull> VectorDomain<AtomDomain<T>> {
-    pub fn new_all() -> Self {
-        Self::new(AtomDomain::<T>::new())
-    }
-}
-impl<D: Domain> Domain for VectorDomain<D> {
-    type Carrier = Vec<D::Carrier>;
-    fn member(&self, val: &Self::Carrier) -> Fallible<bool> {
-        for e in val {
-            if !self.element_domain.member(e)? {
-                return Ok(false);
-            }
-        }
-        Ok(true)
-    }
-}
-
-/// A Domain that specifies the length of the enclosed domain.
-///
-/// # Proof Definition
-/// `SizedDomain(inner_domain, size, D)` is the domain of `inner_domain` restricted to only elements with size `size`.
-///
-/// # Example
-/// First let `inner_domain` be `VectorDomain::new(AtomDomain::<i32>::new())`.
-/// `inner_domain` indicates the set of all i32 vectors.
-///
-/// Then `SizedDomain::new(inner_domain, 3)` indicates the set of all i32 vectors of length 3.
 ///
-/// # Example
+/// # Size Example
 /// ```
-/// use opendp::domains::{SizedDomain, VectorDomain};
-/// // Create a domain that includes all i32 vectors.
-/// let vec_domain = VectorDomain::new_all();
+/// use opendp::domains::{VectorDomain, AtomDomain};
 /// // Create a domain that includes all i32 vectors of length 3.
-/// let sized_domain = SizedDomain::new(vec_domain, 3);
+/// let sized_domain = VectorDomain::new(AtomDomain::default(), Some(3));
 ///
 /// // vec![1, 2, 3] is a member of the sized_domain
 /// use opendp::core::Domain;
```

Comment:
The marker is unnecessary, because the default type of integers in Rust is i32, and this is sufficient for the compiler to infer the type of `AtomDomain::default()` above.

I put it there to make the example easier to read. Although, it would be even easier if it were on the AtomDomain above.